src/virtualenv/create/via_global_ref/builtin/python2/python2.py

```diff
@@ -58,9 +58,11 @@ def sources(cls, interpreter):
         # install files needed to run site.py
         for req in cls.modules():
             stdlib_path = interpreter.stdlib_path("{}.py".format(req))
-            yield PathRefToDest(stdlib_path, dest=cls.to_stdlib)
             comp = stdlib_path.parent / "{}.pyc".format(req)
-            if comp.exists():
+            comp_exists = comp.exists()
+            if stdlib_path.exists() or not comp_exists:
```

Comment:
Why `or not comp_exists`?  In that case, we will generate a broken reference to a file that does not exist, much like the issue this patch (otherwise) addresses.

```
if stdlib_path.exists():
    yield ...
if comp.exists():
    yield ...
```

Seems more correct to me.

If both paths are absent, it may also be worth throwing some kind of exception?  Presumably we cannot possibly succeed without the `req`uirements, so we should error out as soon as we know that.

Reply:
The motivation was to preserve the current behavior when both are missing, mostly because I wasn't sure what to do otherwise. I agree that just checking for each is cleaner.

Reply:
I agree if neither exists we should yield the non compiled version to trigger an error 👍

Reply:
Yielding a file that doesn't exist results in a relatively inscrutable and confusing error message.  Why not just raise an exception here, so it is clear what the fault is?

Reply:
The world is more nuanced than that. The fact that this creator cannot create a virtual environment does not mean that there is not another registered that can. Think e.g. on Python 3 our builtin logic might not be able to create a venv, but the venv creator can by using the interpreters' builtin venv mechanism. Granted if no other creators are found we should probably display an error with the ones that failed due to source invariants could not be satisfied (e.g. missing, permissions, etc). But that's a feature improvement at the creator discovery level, not here.

Reply:
Created https://github.com/pypa/virtualenv/issues/1716 to track the above improvement request.